### ISSUE TITLE
Thank You

### ISSUE THREAD

**Issue by hammypants** [egordorichev/Lens#223]
No issue, just thank you.

If there were to be an issue, it would be that I read that you are thinking of giving up.

You have accomplished so much, so young. Please don't give up.

Thanks again.
-C

**Comment by elliottcarlson** [comment]
Looking forward to digging in to the code -- just bought the Steam copy (others can find it at https://store.steampowered.com/app/851150/Burning_Knight/ )

**Comment by egordorichev** [comment]
Thanks so much, guys! Please do share interesting bits you find in the codebase in this thread!

**Comment by ghost** [comment]
I'll take a look at the code, it looks like a cool game!

**Comment by ucodelukas** [comment]
Awesome game will buy it soon

**Comment by franckyj** [comment]
I'm still trying to figure out where the sword is animated when you boot up the game XD. I saw the sword in the items atlas but where is it animated with the periodic white "shwing!"!?

**Comment by egordorichev** [comment]
@franckyj it does not have a custom swing animation, the sprite itself is rotated with code, and the "white arc" is a separate sprite, that is used for all melee attacks, ([see the ase file](https://github.com/egordorichev/BurningKnight/blob/dev/BurningKnight/Content/Animations/sword_trail.ase)).

**Comment by francky-powerfactors** [comment]
@egordorichev Even the sword that is on the pedestal? before you pick it up, right when you start the game. It oscillates left and right on the pedestal and it looked like a custom animation and not just a rotated sprite.

The black around the sword 'cracks' in ways that I wouldn't expect from a rotated sprite (I would have assumed it would be more of a diagonal line). Maybe I'm just not expressing myself correctly. Oh well!

Really like the game BTW!

**Comment by egordorichev** [comment]
@francky-powerfactors ah, I see, I thought of another animation. Well, any item could be placed on a pedestal, so it would be a suicide mission to animate them all by hand :) It's animated by code, [here is the code for that](https://github.com/egordorichev/BurningKnight/blob/a55594c11ab681087356af2c129c2d493eba4bd2/BurningKnight/entity/item/stand/ItemStand.cs#L238). It's a mix of rotating & moving up, sometimes scaling can be also used.

By 'cracks' you mean it breaks and is not always full? If so, that's the effect of pixel-perfect camera, or rendering at very low resolution and upscaling the result to the full window size. The game viewport is just 320x180. I use some shaders to draw a proper 'full' outline around it, tho, for the white interaction outlines.

I hope this will be useful to you :) Glad to help.

**Comment by francky-powerfactors** [comment]
Seeing bits and pieces of code is helpful but it's a big codebase and I'm not an experienced game developer so it's difficult to grasp all these concepts all at once.

Would it be possible to create a youtube series on how to develop a game similar to this one (albeit way less complex) containing the same feel / mechanics as BurningKnight without all the level of details you put into the real thing? :D :D

**Comment by egordorichev** [comment]
Yeah, it's understandable. Sadly, I'm very busy right now, and I'm pretty bad at talking, but I can link you to some other folks who've explained some concepts I've used. Have a look at [TRASEVOL_DOG](https://twitter.com/TRASEVOL_DOG), especially this video of his :) 

https://www.youtube.com/watch?v=Kut0dirprmU
